shiny/_app.py

```diff
@@ -138,6 +140,17 @@ def _server(inputs: Inputs, outputs: Outputs, session: Session):
 
         self.starlette_app = starlette_app
 
+        if is_uifunc(ui):
+            if iscoroutinefunction(ui):
```

Comment:
I think you'd want to use `_utils.is_async_callable()` here instead of `iscoroutinefunction()`.

But then again, perhaps we should allow async callables? One thing we could do is pass both sync and async functions to `_utils.wrap_async()`. That function always returns an async function, so when we call it later on, we would do `await self.ui(request)`.

That said, I don't foresee a whole lot of use cases for async UI functions, so only allowing sync functions would probably be fine in practice.

Reply:
I'm thinking we'll do async on sort of an as-needed basis, and like you I didn't foresee a lot of use cases for it here.